presto-spark-base/src/test/java/com/facebook/presto/spark/task/TestPrestoSparkTaskExecution.java

```diff
@@ -0,0 +1,137 @@
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.airlift.units.DataSize.Unit.GIGABYTE;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+
+public class TestPrestoSparkTaskExecution
+{
+    Session nativeTestSession;
+    Session nonNativeTestSession;
+    LocalExecutionPlanner.LocalExecutionPlan localExecutionPlan;
+    TaskExecutor taskExecutor;
+    PlanFragment planFragment = createPlanFragment();
+    ExecutorService taskNotificationExecutor;
+    ScheduledExecutorService scheduledExecutor;
+    TaskStateMachine taskStateMachine;
+
+    Set<ScheduledSplit> splits = ImmutableSet.of(
+        new ScheduledSplit(1, TABLE_SCAN_NODE_ID, new Split(new ConnectorId("test"), TestingTransactionHandle.create(), TestingSplit.createLocalSplit())),
+        new ScheduledSplit(2, TABLE_SCAN_NODE_ID, new Split(new ConnectorId("test"), TestingTransactionHandle.create(), TestingSplit.createLocalSplit())),
+        new ScheduledSplit(3, TABLE_SCAN_NODE_ID, new Split(new ConnectorId("test"), TestingTransactionHandle.create(), TestingSplit.createLocalSplit()))
+    );
+
+    @BeforeMethod
+    public void setUp()
+    {
+        taskNotificationExecutor = newCachedThreadPool(daemonThreadsNamed("test-executor-%s"));
+        scheduledExecutor = newScheduledThreadPool(2, daemonThreadsNamed("test-scheduledExecutor-%s"));
+        taskExecutor = new TaskExecutor(8, 16, 3, 4, TASK_FAIR, Ticker.systemTicker());
+
+        nativeTestSession = testSessionBuilder()
+                .setSystemProperty(NATIVE_EXECUTION_ENABLED, "true")
+                .setCatalog("tpch")
+                .setSchema(TINY_SCHEMA_NAME)
+                .build();
+        nonNativeTestSession = testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema(TINY_SCHEMA_NAME)
+                .build();
+
+        taskStateMachine = new TaskStateMachine(new TaskId("test_query_id", 4, 4, 1), taskNotificationExecutor);
+
+        localExecutionPlan = createTestingPlanner().plan(
+                TestingTaskContext.createTaskContext(taskNotificationExecutor, scheduledExecutor, nativeTestSession),
+                planFragment,
+                new TestingOutputBuffer(),
+                new TestingRemoteSourceFactory(),
+                new TableWriteInfo(Optional.empty(), Optional.empty(), Optional.empty()),
+                new ArrayList<>());
+    }
+
+    @AfterMethod
+    public void tearDown()
+    {
+        taskStateMachine.finished();
+        taskExecutor.stop();
+        scheduledExecutor.shutdown();
+        taskNotificationExecutor.shutdown();
+    }
+
+    @Test
+    public void testNativeSourceNodeDriverSplitRunnerInstanceCount()
```

Comment:
These names are super long and hard to read. Any particular reason not to shorten them to something like testNativeDriverCount and testJavaDriverCount?